### PR TITLE
[PoC] surface Quic transport errors

### PR DESCRIPTION
**Pull request by wfurt**
This implements  proposal from #72666.

We would currently surface TransportError on cipher mismatch. I suspect that both MsQuic and other implementations may have more codes that we do not handle so the actual set may be bigger.

Looking at the cases again when we throw on transport error, I would like to prose change on QUIC_STATUS_USER_CANCELED.

While that maps to MsQuic status, I feel cancellation has specific meaning in .NET around cancellation tokens. 
In tests where this happens there is no cancellation nor user. Typically this is failure to complete handshake operation. (like callback failure) 

```diff
-  System.Security.Authentication.AuthenticationException: Authentication failed because the remote party sent a TLS alert: 'UserCanceled'.
+  System.Net.Quic.QuicException: QUIC encountered transport error '346'
```

I also feel that when we would surface as transport error, we would be more following RFC than MsQuic internals. In either case there is not much the code can do e.g. this is purely diagnostic.

### CODE REVIEW

File: src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicCipherSuitesPolicyTests.cs [hunk 1]
```diff
@@ -71,10 +71,12 @@ public void NoSupportedCiphers_ThrowsArgumentException(TlsCipherSuite[] ciphers)
         [Fact]
         public async Task MismatchedCipherPolicies_ConnectAsync_ThrowsQuicException()
         {
-            await Assert.ThrowsAsync<QuicException>(() => TestConnection(
+            QuicException ex = await Assert.ThrowsAsync<QuicException>(() => TestConnection(
```
Comment (rzikm) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicCipherSuitesPolicyTests.cs:74]: `There is a helper for this, something like `AssertThrowsQuicException(QuicError.TransportError, () => ...);`

**Comment by ManickaP** [comment]
I'm not a fan of this particular proposition:
```diff
-  System.Security.Authentication.AuthenticationException: Authentication failed because the remote party sent a TLS alert: 'UserCanceled'.
+  System.Net.Quic.QuicException: QUIC encountered transport error '346'
```

I don't think making exception more obscure is better. I don't find the combination of TLS alert and cancel misleading, but if you find it unsuitable, is there anything better we could replace this with?

File: src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicCipherSuitesPolicyTests.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicCipherSuitesPolicyTests.cs]: updated.

**Comment by ManickaP** [comment]
This part of `QuicStream` code is also relevant in this change:
https://github.com/dotnet/runtime/blob/2c62994efb2495dcaef2312de3ab25ea4792b23a/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs#L583-L590

**Comment by wfurt** [comment]
closing in favor of  #88550

**Pull request closed without merging**